tests/main_test.go

```diff
@@ -0,0 +1,1078 @@
+package main_test
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/joho/godotenv"
+)
+
+func TestSingleLabelRebuild(t *testing.T) {
+
+	envMap, mapErr := getEnvMap()
+	if mapErr != nil {
+		t.Fatalf("%v", mapErr)
+	}
+	hosts := []string{"ncn-w001"}
+	var rebuildResponse RebuildResponse
+
+	err := rebuildHosts(envMap["REBUILD_URL"], hosts, &rebuildResponse)
+
+	if err != nil {
+		t.Fatalf("could not rebuild hosts: %v", err.Error())
+	}
+
+	// wait for workflow to start
+	// DI: let me know if theres a smarter way to handle this
+	time.Sleep(5 * time.Second)
+
+	var getResponse GetResponse
+	getRebuildStatus(envMap["STATUS_URL"], "target-ncns="+hosts[0], &getResponse)
+
+	for getResponse[0].Status.Phase == "Running" {
+		// make get request to check status
+		// TODO: handle error that this returns
+		getRebuildStatus(envMap["STATUS_URL"], "target-ncns="+hosts[0], &getResponse)
+
+		// DI: Let me know if you would like me to sleep here or just request as many times as possible?
+		// time.Sleep(2 * time.Second)
+
+	}
+
+	// TODO: Fail here in more cases
+	if getResponse[0].Status.Phase != "Succeeded" {
+		t.Fatalf("Expected phase to be Succeeded but got: %v", getResponse[0].Status.Phase)
+
+	}
+
+}
+
+func rebuildHosts(url string, hosts []string, target interface{}) error {
+
+	hoststostring := ``
+
+	for i := 0; i < len(hosts); i++ {
+		if i == len(hosts)-1 {
+			hoststostring += `"` + hosts[i] + `"`
+		} else {
+			hoststostring += `"` + hosts[i] + `",`
+		}
+	}
+
+	requestBody := strings.NewReader(`
+		{
+		"dryRun": true,
+		"hosts": [
+		  ` + hoststostring + `
+		]
+	  	}`)
+
+	response, err := http.Post(url, "application/json", requestBody)
+	defer response.Body.Close()
+	if err != nil {
+		return errors.New("could not complete POST request: " + err.Error())
+	}
+
+	// content, _ := ioutil.ReadAll(response.Body)
+
+	return json.NewDecoder(response.Body).Decode(target)
+
+}
+
+func getEnvMap() (map[string]string, error) {
+
+	err := godotenv.Load(".env")
+	if err != nil {
+		return nil, errors.New("could not load .env file")
+	}
+	envMap, mapErr := godotenv.Read(".env")
+	if mapErr != nil {
+		return nil, errors.New("could not read .env file")
+	}
+	return envMap, nil
+}
+
+func getRebuildStatus(url string, label string, target interface{}) error {
+
+	url += "?labelSelector=" + label
+
+	response, err := http.Get(url)
+	defer response.Body.Close()
+	if err != nil {
+		return err
+	}
+
+	return json.NewDecoder(response.Body).Decode(target)
+}
+
+type RebuildResponse struct {
+	Name       string   `json:"name"`
+	TargetNcns []string `json:"targetNcns"`
+}
+
+type GetResponse []struct {
```

Comment:
this feels wrong to me. it sounds like this is a struct of one build. I think you can use argo types instead of implementing your own